data/set/basic.lean

```diff
@@ -823,6 +823,10 @@ theorem image_inter {f : α → β} {s t : set α} (H : injective f) :
   f '' s ∩ f '' t = f '' (s ∩ t) :=
 image_inter_on (assume x _ y _ h, H h)
 
+theorem surjective_onto_image {f : α → β} {s : set α} :
+  surjective (λ p, ⟨f p.1, ⟨p.1, p.2, rfl⟩⟩ : s → f '' s) :=
```

Comment:
It would be nicer to use the lemma `mem_image_of_mem` (like `mem_range_self`) in the later proof.

Reply:
I think this function should be given a name, so that the proofs don't appear in the statement of the theorem.